fhir-persistence/src/test/java/com/ibm/fhir/persistence/test/common/AbstractReverseChainTest.java

```diff
@@ -0,0 +1,363 @@
+        List<Resource> resources = runQueryTest(Encounter.class, queryParms);
+        assertNotNull(resources);
+        assertEquals(1, resources.size());
+        assertEquals("Encounter", resources.get(0).getClass().getSimpleName());
+        assertEquals(savedEncounter1.getId(), resources.get(0).getId());
+    }
+
+    /**
+     * This test queries for Encounters which are referenced by Observations with a
+     * reference to Patients with a specified name.
+     * One patient is found, thus one observation, thus one encounter is returned.
+     * @throws Exception
+     */
+    @Test
+    public void testReverseChainSingleResultWithChainedParm() throws Exception {
+        Map<String, List<String>> queryParms = new HashMap<String, List<String>>();
+        queryParms.put("_has:Observation:encounter:subject:Patient.name", Collections.singletonList("Vito"));
+        List<Resource> resources = runQueryTest(Encounter.class, queryParms);
+        assertNotNull(resources);
+        assertEquals(1, resources.size());
+        assertEquals("Encounter", resources.get(0).getClass().getSimpleName());
+        assertEquals(savedEncounter1.getId(), resources.get(0).getId());
+    }
+
+    /**
+     * This test queries for Encounters which are referenced by Observations
+     * matching on a string type search parameter.
+     * One observation is found, thus one encounter is returned.
+     * @throws Exception
+     */
+    @Test
+    public void testReverseChainSingleResultWithStringParm() throws Exception {
+        Map<String, List<String>> queryParms = new HashMap<String, List<String>>();
+        queryParms.put("_has:Observation:encounter:value-string", Collections.singletonList("test"));
+        List<Resource> resources = runQueryTest(Encounter.class, queryParms);
+        assertNotNull(resources);
+        assertEquals(1, resources.size());
+        assertEquals("Encounter", resources.get(0).getClass().getSimpleName());
+        assertEquals(savedEncounter1.getId(), resources.get(0).getId());
+    }
+
+    /**
+     * This test queries for Encounters which are referenced by Observations
+     * matching on a composite type search parameter.
+     * One observation is found, thus one encounter is returned.
+     * @throws Exception
+     */
+//    @Test
+//    public void testReverseChainSingleResultWithCompositeParm() throws Exception {
+//        Map<String, List<String>> queryParms = new HashMap<String, List<String>>();
+//        queryParms.put("_has:Observation:encounter:code-value-string", Collections.singletonList("code$test"));
+//         List<Resource> resources = runQueryTest(Encounter.class, queryParms);
+//        assertNotNull(resources);
+//        assertEquals(1, resources.size());
+//        assertEquals("Encounter", resources.get(0).getClass().getSimpleName());
+//        assertEquals(savedEncounter1.getId(), resources.get(0).getId());
+//    }
```

Comment:
Maybe comment why this commented out?

Reply:
added TODO indicating this should be uncommented when issue #1669 is resolved